ex04.ex

```diff
@@ -75,7 +75,20 @@ defmodule Ex04 do
   return value will be the thing you have to manipulate.
   """
 
-  def even_odd . . . "your code"
+    #helper function used to reverse the two lists in a tuple
```

Comment:
### -2

There are a couple of problems here. First, the `even_odd` function is a bit long. Second, the use of `true` in the `cond` means that you'll accept a list of strings and say they're all odd.

~~~ elixir
  def even_odd(list) do
    reduce(list, { [], [] }, &partition_even_odd/2)
    |> reverse_two_lists
  end

  defp partition_even_odd(x, { es, os }) when Integer.is_even(x), do: {[x | es], os}
  defp partition_even_odd(x, { es, os }) when Integer.is_odd(x),  do: {es, [x | os]}

  defp reverse_two_lists({ evens, odds }) do
    { reverse(evens), reverse(odds) }
  end
~~~